azurerm/internal/services/sql/helper/sqlExtendedAuditing.go

```diff
@@ -0,0 +1,99 @@
+package helper
+
+import (
+	"github.com/Azure/azure-sdk-for-go/services/preview/sql/mgmt/2017-03-01-preview/sql"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func ExtendedAuditingSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeList,
+		Optional: true,
+		MaxItems: 1,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"storage_account_access_key": {
+					Type:         schema.TypeString,
+					Required:     true,
+					Sensitive:    true,
+					ValidateFunc: validate.NoEmptyStrings,
+				},
+
+				"storage_endpoint": {
+					Type:         schema.TypeString,
+					Required:     true,
+					ValidateFunc: validate.URLIsHTTPS,
+				},
+
+				"storage_account_access_key_is_secondary": {
```

Comment:
could we order these alphabetically

Reply:
The same concern. So we don't need to put required fields in advance? We just order all required and optional fields alphabetically, right?